AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine.swift

```diff
@@ -160,66 +160,61 @@ final class StorageEngine: StorageEngineBehavior {
 
     func delete<M: Model>(_ modelType: M.Type,
                           withId id: Model.Identifier,
-                          completion: @escaping (DataStoreResult<Void>) -> Void) {
-        let wrappedCompletion: DataStoreCallback<Void> = { result in
-            guard modelType.schema.isSyncable, let syncEngine = self.syncEngine else {
-                if !modelType.schema.isSystem {
-                    self.log.error("Unable to sync modelType (\(modelType)) where isSyncable is false")
-                }
-                if self.syncEngine == nil {
-                    self.log.error("Unable to sync because syncEngine is nil")
-                }
-                completion(result)
+                          completion: @escaping (DataStoreResult<M?>) -> Void) {
+        let transactionResult = delete(modelType, predicate: field("id").eq(id))
```

Comment:
Why did we start using the predicate version for deleting withId?  If we can avoid using the predicate version, we should to avoid having to do a transaction?

Reply:
this consolidated the code paths for all Delete APIs to a single delete method, though there might be a performance difference between using the predicate verse using the Id based on the SQL statement being constructed, lemme take a look at this-

sounds right about not needing a transaction for a single query+delete

Reply:
so it looks like the query(byId) and delete(byId) are both creating predicates which create conditional sql statements so there is no difference between using the predicate version.

Reply:
synced on this; the atomicity that we are looking for is to ensure an external update like DataStore.save does not happen between the entire operation of the delete API, ie. between the sql query and sql delete. If it does happen without the transaction, then the subscribers to the model will get an update that the model was deleted with the fields from the query, and not the actual field values that were modified by the update